core/include/detray/materials/material.hpp

```diff
@@ -91,12 +112,23 @@ struct material {
         return m_z * m_molar_rho;
     }
 
+    /// @returns the density effect data
+    DETRAY_HOST_DEVICE
+    constexpr detail::density_effect_data<scalar_type> density_effect_data()
```

Comment:
Is this supposed to copy the densitiy effect data?
```suggestion
    constexpr const detail::density_effect_data<scalar_type>& density_effect_data()
```

Reply:
Right. Fixed now